<!-- Launches lidar, hector slam, camera pub/edit, and marker pub -->
<!-- Basically same as:
        roslaunch autobot hector_slam.launch (lidar, hector slam, rviz)
        roslaunch autobot camera.launch (camera_pub, camera_edit, rqt)
        rosrun autobot markers_pub.py
        rviz
-->
<!-- To run: roslaunch autobot viz_markers_with_slam.launch -->

<launch>

  <!-- launch lidar -->
  <node name="ydlidar_node" pkg="ydlidar" type="ydlidar_node" output="screen" respawn="false" >
      <param name="port" type="string" value="/dev/ydlidar"/>
      <param name="baudrate" type="int" value="128000"/>
      <param name="frame_id" type="string" value="laser_frame"/>
      <param name="low_exposure" type="bool" value="false"/>
      <param name="resolution_fixed" type="bool" value="true"/>
      <param name="auto_reconnect" type="bool" value="true"/>
      <param name="reversion" type="bool" value="false"/>
      <param name="angle_min" type="double" value="-180" />
      <param name="angle_max" type="double" value="180" />
      <param name="range_min" type="double" value="0.1" />
      <param name="range_max" type="double" value="16.0" />
      <param name="ignore_array" type="string" value="" />
      <param name="samp_rate" type="int" value="9"/>
      <param name="frequency" type="double" value="7"/>
  </node>

  <!-- launch hector slam -->
  <node pkg ="tf" type="static_transform_publisher" name="map_to_odom" args="0.0 0.0 0.0 0.0 0.0 0.0 /map /nav 40"/>
  <node pkg ="tf" type="static_transform_publisher" name="odom_to_base_link" args="0.0 0.0 0.0 0.0 0.0 0.0 /nav /base_link 40"/>
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser" args="0.2245 0.0 0.2 0.0 0.0 0.0 /base_link /laser_frame 40" />
  <include file="$(find hector_mapping)/launch/mapping_default.launch" />
  <node pkg="rviz" name="rviz" type="rviz" args="-d $(find autobot)/rviz/viz_markers_with_slam.rviz" />
  <include file="$(find hector_geotiff)/launch/geotiff_mapper.launch" />

    <!-- launch camera pub, camera editor, and rqt -->
    <include file="camera.launch"/>

    <!-- publish cylinder center point -->
    <node pkg="autobot" name="markers_pub" type="markers_pub.py" output="screen"/>

</launch>